### DISCUSSION TITLE
How set some rows as initially selected

### DISCUSSION THREAD

**Discussion by dlauterbach** [TanStack/table#2435] [General] [answered]
I am displaying a table with row selection in a modal.  When the user clicks "Save", the selected rows are saved into the parent component state and the modal is closed.  If the user later clicks on button to edit their selections in same modal, I would like to display the table with their previously selected rows from parent component state initially selected.  How can I load a table with some rows initially selected?

**Comment by dlauterbach** [comment, marked as answer]
I found my answer [here](https://github.com/tannerlinsley/react-table/blob/master/docs/api/useRowSelect.md#table-options).  To have some rows initially selected I add their row Ids to initialState.selectedRowIds object that is passed as option to useTable(options).

**Reply by tahv0** [reply]
Your link is broken. Any chance to get correct link?

**Reply by aquibbaig** [reply]
https://github.com/tannerlinsley/react-table/blob/master/docs/src/pages/docs/api/useRowSelect.md

**Reply by AmrStar25** [reply]
but this still have issue because table will re-render after each change in row selection and initialState.selectedRowIds remain as it and i cant't change any row selection @dlauterbach

**Reply by Lokinorse** [reply]
> https://github.com/tannerlinsley/react-table/blob/master/docs/src/pages/docs/api/useRowSelect.md

Your link is broken. Any chance to get correct link?

**Reply by dhruv-pixis** [reply]
Can you provide the full solution here ?
@dlauterbach

**Reply by dhruv-pixis** [reply]
Did you find any solution for this ? @Lokinorse

**Comment by GuOshiro** [comment]
Hey @tahv0 you can see here 
https://spectrum.chat/react-table/general/v7-how-get-selected-rows-outside-of-react-table~2deb9558-c484-4b97-9e1c-6be608f1f275

**Reply by Lokinorse** [reply]
Your link is broken. Any chance to get correct link?

**Comment by ialexalvarado** [comment]
any new URL for this topic? that URL's already broken

**Reply by codeawy** [reply]
You can use resetRowSelection() to reset, I have it help

**Comment by devGemScram** [comment]
Please provide a working URL, the above URL is broken.

**Reply by codeawy** [reply]
You can use resetRowSelection() to reset, I have it help

**Reply by Lokinorse** [reply]
Initial question was: "rows as initially selected". How exactly resetRowSelection() will help?

**Comment by QastanFIT** [comment]
Anyone with a working example?

**Reply by dubbajubba** [reply]
You need to define your `rowSelection` which is an object of records where the key is the row's ID and the value is a `boolean`. An example row selection would look like this:
```tsx
{
   0: true,
   1: false,
}
```

Alternatively, you could omit the false items like this:
```tsx
{
   0: true,
}
```

Now, to set the initial state, you would pass the row selection to the initialState like:
```tsx
const [rowSelection, setRowSelection] = useState({});
const table = useReactTable({
   // the rest of your configuration
   state: {
      rowSelection
   }
   initialState: {
      rowSelection
   }
});
```

Please note that the `rowSelection` is being used in `state` and `initialState`, I generally only set the row selection in `state` because I am generally fetching the data and setting the row selection after fetching data, so the initial state doesn't apply to most of my cases. If you know the selected rows before rendering your table, you can set it in `initialState`.

One more note, if you are using the record's `id`, not the row index, the key in the rowSelection record would be the `id` of the resource, not the row's index. (e.g., when setting your row id like this: `getRowId: (row) => row.id`).

You can read more info here: [https://tanstack.com/table/v8/docs/guide/row-selection#manage-row-selection-state](https://tanstack.com/table/v8/docs/guide/row-selection#manage-row-selection-state)

**Reply by tungrix** [reply]
When I set:
```javascript
const [rowSelection, setRowSelection] = useState({"1": true});
```
I got:
```typescript
Type 'Dispatch<SetStateAction<{ "1": boolean; }>>' is not assignable to type 'OnChangeFn<RowSelectionState>'.
  Types of parameters 'value' and 'updaterOrValue' are incompatible.
    Type 'Updater<RowSelectionState>' is not assignable to type 'SetStateAction<{ "1": boolean; }>'.
      Type 'RowSelectionState' is not assignable to type 'SetStateAction<{ "1": boolean; }>'.
        Property '"1"' is missing in type 'RowSelectionState' but required in type '{ "1": boolean; }'.
```
from `onRowSelectionChange` inside `useReactTable`.

**Reply by KevinVandy** [reply]
Use strings for the keys

**Reply by tungrix** [reply]
I changed to string and still got an error from typescript, edited the error above.

**Reply by dubbajubba** [reply]
I think typescript is inferring the type of rowSelection as `Record<"1", boolean>`.

Type your variable, like this: `const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({ "1": true });`

**Reply by tungrix** [reply]
Thank you, adding `<RowSelectionState>` to the useState also works.
```typescript
import { RowSelectionState } from "@tanstack/react-table";

const [rowSelection, setRowSelection] = useState<RowSelectionState>({"1": true});

```

**Reply by dhruv-pixis** [reply]
This works.

Mark this as an `ANSWER`
